ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -84,26 +85,42 @@ public RequestValidator(
    * @param sql                 the sql that generated the list of statements, used for
    *                            generating more useful debugging information
    *
-   * @return the number of new persistent queries that would be created by {@code statements}
+   * @return the offset that the statements were validated against
    * @throws KsqlException if any of the statements cannot be validated, or the number
    *                       of requested statements would cause the execution context
    *                       to exceed the number of persistent queries that it was configured
    *                       to support
    */
-  public int validate(
+  public long validate(
       final List<ParsedStatement> statements,
       final Map<String, Object> propertyOverrides,
       final String sql
   ) {
-    final KsqlExecutionContext ctx = snapshotSupplier.get();
+    final EngineSnapshot snapshot = snapshotSupplier.get();
+    final KsqlExecutionContext ctx = snapshot.getContext();
     final SchemaInjector injector = schemaInjectorFactory.apply(serviceContext);
 
+    validate(ctx, injector, statements, propertyOverrides, sql);
+
+    return snapshot.getOffset();
+  }
+
+  /*
+   * @return the number of new persistent queries that would be created by {@code statements}
+   */
+  private int validate(
```

Comment:
is the return value used? I think ti used to be checked to ensure query quotas weren't exceeded....